### ISSUE TITLE
bad result with multi-speaker MB-melgan

### ISSUE THREAD

**Issue by zhouyong64** [TensorSpeech/TensorFlowTTS#511]
I tried the baker example with mb-melgan and it works great. But when I add data from two other speakers to the baker speaker to make it a 3-person multi-speaker dataset, me-melgan's train and eval losses become pretty bad. 

train_gen_loss:
021-03-01 21:03:40,095 (base_trainer:658) INFO: (Step: 2200) train_gen_loss = 2.5342.
2021-03-01 21:04:23,791 (base_trainer:658) INFO: (Step: 2400) train_gen_loss = 2.6793.
2021-03-01 21:05:07,897 (base_trainer:658) INFO: (Step: 2600) train_gen_loss = 1281.3123.
2021-03-01 21:05:51,570 (base_trainer:658) INFO: (Step: 2800) train_gen_loss = 1442.6925.
...
2021-03-02 09:55:21,211 (base_trainer:658) INFO: (Step: 199600) train_gen_loss = 1476.6885.
2021-03-02 09:56:05,495 (base_trainer:658) INFO: (Step: 199800) train_gen_loss = 1436.1117.
2021-03-02 09:56:49,795 (base_trainer:658) INFO: (Step: 200000) train_gen_loss = 1466.0608.
 
eval_gen_loss:
2021-03-01 21:15:07,627 (base_trainer:527) INFO: (Steps: 5000) eval_gen_loss = 914.3601.
2021-03-01 21:34:31,095 (base_trainer:527) INFO: (Steps: 10000) eval_gen_loss = 909.6656.
2021-03-01 21:53:55,125 (base_trainer:527) INFO: (Steps: 15000) eval_gen_loss = 909.4446.
...
2021-03-02 09:18:52,724 (base_trainer:527) INFO: (Steps: 190000) eval_gen_loss = 984.7055.
2021-03-02 09:38:28,286 (base_trainer:527) INFO: (Steps: 195000) eval_gen_loss = 945.8341.
2021-03-02 09:58:05,141 (base_trainer:527) INFO: (Steps: 200000) eval_gen_loss = 996.6537.
The above two are "training generator with only stft loss". Resuming train with generator + discriminator won't reduce the losses much.

**Comment by dathudeptrai** [comment]
@zhouyong64 you should check ur dataset and remove bad audio (almost silence, noise ...), this problem is almost caused by a dataset.

**Comment by zhouyong64** [comment]
You're right. If one of the speakers is removed, the result will be better. However, the same 3-speaker dataset works OK with WaveRNN, which seems more robust than mb-melgan.

**Comment by dathudeptrai** [comment]
> You're right. If one of the speakers is removed, the result will be better. However, the same 3-speaker dataset works OK with WaveRNN, which seems more robust than mb-melgan.

yes, mb-melgan is more sensitive with bad samples :(. But you can easily fix it with masked loss (pls refer https://github.com/TensorSpeech/TensorFlowTTS/issues/247) @zhouyong64

**Comment by zhouyong64** [comment]
Great, I'll try. By the way, would you consider adding a universal vocoder such as universal melgan to this repo? I think it would be a great addition, eliminating the need to train on multi-speaker data in some cases. In my experience, melgan and mb-melgan don't generalize well to unseen speakers, while WaveRNN seems a little better with respect to generalization.

**Comment by dathudeptrai** [comment]
@zhouyong64 you can refer (https://github.com/TensorSpeech/TensorFlowTTS/issues/506) for the implementation of universal melgan. If you can make it work, pls create a PR for this :D.

**Comment by zhouyong64** [comment]
I'll take a look. Another thing, the masked loss trick you just mentioned, have you tried it with other models? It seems a great trick to deal with low-quality data if it can be applied to other models as well.

**Comment by dathudeptrai** [comment]
> I'll take a look. Another thing, the masked loss trick you just mentioned, have you tried it with other models? It seems a great trick to deal with low-quality data if it can be applied to other models as well.

yes, you can apply it for other models but need tune the threshold value by urself :D.

**Comment by zhouyong64** [comment]
OK

**Comment by ymzlygw** [comment]
@zhouyong64  Hi，Can you tell me the waveRNN you try is based on Pytorch or TF? I want to find a TF2 impementation of waveRNN.Thanks!
